008-strimzi-ui.md

```diff
@@ -0,0 +1,119 @@
+# Add a UI to Strimzi for Kafka Administration
+
+This proposal is to add a Kafka administration UI to the Strimzi project.
+
+## Motivation
+
+Kafka out of the box provides a number of useful command tools to perform common administrative tasks such as creating topics, viewing consumer group status, and consuming record content. While these are helpful, they are not always the simplest or most convenient way of monitoring and performing administrative tasks on a Kafka cluster. In these cases a dedicated administration UI, focused on common administration tasks, would not only enable a user to complete their task more quickly and simply than previously possible, but also enable a more positive user experience of Kafka and its capabilities.
+
+## Proposal
+
+This proposal will describe a Kafka administration UI capability. It will describe example capabilities which could be offered, cover at a high level how the UI code itself could be implemented (with more low level detail to be provided as required), how this could be delivered, as well as possible future work and how this could integrate with Strimzi today.
+
+### UI implementation
+
+At a high level, I would propose a UI implemented as follows:
+
+- It would be a [React](https://reactjs.org/) based UI
+- This UI is hosted and provided via an [Express](https://expressjs.com/) server
+- This UI is built in a modular/metadata driven manner - allowing for easy extensibility, modification, and dynamic behaviour at runtime
+- Uses a [GraphQL](https://graphql.org/) API
+- Should be fully translatable, and accessible
+
+I would suggest that the UI implementation to follow the Model View Controller (MVC) pattern, with clear separation between business logic, and the view logic that renders the state of the UI. This would enable/allow the view logic to be modified/swapped in/out as required, but keep the same business logic. I would suggest using the [Carbon](https://www.carbondesignsystem.com/) design system as the view layer for the contributed UI, given it is open source, is supported, and I can provide both the design and implementation using it. As an example, the following is a mock up of how a Topics page could look using the Carbon design system, which would allow a user to view, create, edit and delete topics in a given cluster:
```

Comment:
Since you've added so many links to other technologies and concepts, is it worth throwing one in for [MVC](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller) or is it widely known enough not to merit that?

Reply:
Good point! I will add one (the Wikipedia description does a nice job of introducing it)